Feature/school 13

Flictivity opened the pull request.

Flictivity committed:
SCHOOL-13
Создана страница с кружками

WingiM committed:
SCHOOL-12
Добавлен базовый класс для sql-getters

WingiM committed:
Merge branch 'dev' into feature/SCHOOL-12

# Conflicts:
#	DumaemSchool.BlazorWeb/Pages/FetchData.razor
#	DumaemSchool.Database/ListGetters/Impl/TeacherListSqlGenerator.cs

WingiM committed:
SCHOOL-12
Переработан процесс получения условий выборки

WingiM committed:
SCHOOL-12
Добавлен запрос на получение информации о кружках
Добавлен запрос на получение информации об учителе

Flictivity committed:
Merge remote-tracking branch 'origin/dev' into feature/SCHOOL-13

Flictivity committed:
Merge remote-tracking branch 'origin/feature/SCHOOL-12' into feature/SCHOOL-13

Flictivity committed:
SCHOOL-13
Добавлен конвертер фильтров

Flictivity committed:
SCHOOL-13
Реализована страница кружков (отображение)
Добавлен компонент для отображения кружков

WingiM commented on DumaemSchool.BlazorWeb/Data/Converters/FilterConverter.cs:
```diff
@@ -0,0 +1,53 @@
+{
+    private static readonly Dictionary<string, FilterOperand> Operands = new()
+    {
+        {"contains", FilterOperand.Contains},
```
убрать хардкод

WingiM commented on DumaemSchool.BlazorWeb/Shared/MainLayout.razor:
```diff
@@ -14,12 +14,12 @@
                     <MudIconButton Size="Size.Medium" Icon="@Icons.Material.Filled.Category" Color="Color.Inherit" />
                 </MudTooltip>
-                <MudTooltip Text="Учителя">
+                <MudTooltip Text="Учтеля">
```
чт

WingiM commented on DumaemSchool.BlazorWeb/Pages/SectionsPage.razor:
```diff
@@ -0,0 +1,8 @@
+@using MediatR
+@using DumaemSchool.BlazorWeb.Data.Converters
+
+<Sections TeacherId="1"></Sections>
```
добавить код проверки авторизованного пользователя

WingiM commented on DumaemSchool.BlazorWeb/Data/Converters/FilterConverter.cs:
```diff
@@ -0,0 +1,53 @@
+
+namespace DumaemSchool.BlazorWeb.Data.Converters;
+
+public static class FilterConverter<T>
```
женерик убрать попробуй

Flictivity committed:
SCHOOL-13
Изменен конвертер фильтра (больше не Generic и изменен словарь)
Добавлена обработка роли пользователя в компоненте со списком кружков

WingiM commented on DumaemSchool.BlazorWeb/appsettings.json:
```diff
@@ -6,7 +6,7 @@
   },
   "ConnectionStrings": {
-    "Default": "Host=localhost;Port=5432;Database=dumaem_school;Username=postgres;Password=postgres;"
+    "Default": "Host=localhost;Port=6432;Database=dumaem_school;Username=postgres;Password=rinatSokrat123;"
```
откати и поменяй пароль

Flictivity committed:
SCHOOL-13
Изменен пароль БД
